### PR TITLE
Preset: require spaces in anonumous fe for crockford style

### LINKED ISSUE #1351: Preset definition for 'crockford' coding style should include requireSpacesInAnonymousFunctionExpression
From http://javascript.crockford.com/code.html:
`If a function literal is anonymous, there should be one space between the word function and the ( (left parenthesis).`

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/13606887-preset-definition-for-crockford-coding-style-should-include-requirespacesinanonymousfunctionexpression?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by markelog**
Fixes #1351

### CODE REVIEW

File: test/specs/config/generator.js [hunk 1]
```diff
@@ -17,7 +17,8 @@ describe('lib/config/generator', function() {
         '\u001b[32mdisallowDanglingUnderscores\u001b[39m (25 violations in 1 file):\n    Create an (e)xception for this rule, or (f)ix the errors yourself?': 'e',
         '\u001b[32mrequireSpaceAfterKeywords\u001b[39m (17 violations in 1 file):\n    Create an (e)xception for this rule, or (f)ix the errors yourself?': 'e',
         '\u001b[32mrequireBlocksOnNewline\u001b[39m (2 violations in 1 file):\n    Create an (e)xception for this rule, or (f)ix the errors yourself?': 'e',
-        '\u001b[32mdisallowSpaceBeforeBinaryOperators\u001b[39m (1 violation in 1 file):\n    Create an (e)xception for this rule, or (f)ix the errors yourself?': 'e'
+        '\u001b[32mdisallowSpaceBeforeBinaryOperators\u001b[39m (1 violation in 1 file):\n    Create an (e)xception for this rule, or (f)ix the errors yourself?': 'e',
+        '\u001b[32mrequireSpacesInAnonymousFunctionExpression\u001b[39m (17 violations in 1 file):\n    Create an (e)xception for this rule, or (f)ix the errors yourself?': 'e'
```
Comment (markelog) [inline test/specs/config/generator.js]: A sad test dependency :(

/cc @mrjoelkemp

Comment (mrjoelkemp) [inline test/specs/config/generator.js]: Which test fails when you exclude this? Not seeing why this is necessary.

Comment (mrjoelkemp) [inline test/specs/config/generator.js]: Also didnt have to add anything to expectedConfig which is a tad confusing.

**Comment by mrjoelkemp** [comment]
+1 except for the autoconfig test confusion

File: test/specs/config/generator.js (hunk 1 above)
Comment (markelog) [inline test/specs/config/generator.js]: > Which test fails when you exclude this?

See https://github.com/jscs-dev/node-jscs/pull/1357/files#diff-8c69b27e094f551b0135c745b7940c78R117

Comment (mrjoelkemp) [inline test/specs/config/generator.js]: Not immediately seeing why this happens. Can you land this and add a card to remove this dependency? I'll tackle it soon.

Comment (markelog) [inline test/specs/config/generator.js]: Sure, to remove this dep we could simple register new preset for the test

Comment (mrjoelkemp) [inline test/specs/config/generator.js]: Yes, that would be much better.

**Pull request closed without merging**